src/reconcile/rec_visibility.c

```diff
@@ -798,9 +830,16 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
      * Paranoia: check that we didn't choose an update that has since been rolled back.
      */
     WT_ASSERT(session, upd_select->upd == NULL || upd_select->upd->txnid != WT_TXN_ABORTED);
-    /* We should never select an update that has been written to the history store. */
-    WT_ASSERT(session, upd_select->upd == NULL || !F_ISSET(upd_select->upd, WT_UPDATE_HS));
-    WT_ASSERT(session, tombstone == NULL || !F_ISSET(tombstone, WT_UPDATE_HS));
+    /*
+     * We should never select an update that has been written to the history store except checkpoint
+     * writes the update that is older than a prepared update.
+     */
+    WT_ASSERT(session,
```

Comment:
In this scenario, the same update is present in both data and history store. Moving this update again to the history store can trigger the OOO fix.

Reply:
No before we move it, we first delete it from the history store.

Reply:
No, it has the hs flag so will not be moved to hs again.

Reply:
Yeah, understood, even in rollback, we set the flag to delete them before reinsert.